streams/shared/src/main/scala/scalaz/stream/ZStream.scala

```diff
@@ -660,6 +660,16 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
       }
   }
+
+  /**
+   * @note in case you need extreme performance, make sure to use capacity
+   *   which is a power of 2.
+   * Stores this stream to a fixed managed queue and streams data from the managed queue.
+   */
+  final def buffer[R1 <: R, E1 >: E, A1 >: A](capacity: Int): ZManaged[R, Nothing, ZStream[R, E, A]] =
```

Comment:
Let's use `Stream.managed` and `flatten` to unwrap the `ZManaged` here.

Reply:
Oh and I believe the type parameters here are unneeded.

Reply:
Fixed